errors/list.go

```diff
@@ -102,14 +102,13 @@ func (l *List) Detailed() string {
 	return strings.Join(details, "\n")
 }
 
-// Append appends a new error on the error list.
-// If the error is nil it will not be added on the error list.
+// Append appends the provided errs on the error list, ignoring nil values.
 //
-// Any error of type hcl.Diagnostics will be flattened inside
-// the error list, each hcl.Diagnostic will become an error.Error.
+// Any error of type hcl.Diagnostics will have its hcl.Diagnostic elements added
+// to the error list.
 //
-// Any error of type errors.List will be flattened inside
-// the error list.
+// Any error of type *List will have its error elements added to the
```

Comment:
hmm for me it was quite clear that if I append a list on another list I would end up having a list of lists ( [ [err1, err2] ] ) and to avoid that and have a single list of errors ([ err1, err2]) what need to happen is flattening, so you always have a single list of errors. So not entirely sure how the new phrasing is better 🤔

Reply:
ok, it's because for me "flatten" means something different depending on the case... I just wanted to be explicit about what was happening here but probably the wording was good enough. will rollback.

Reply:
hmm I could be wrong on my take on this, what I get by flattening is this:

```
a = [1,2,3]
a.append([4,5])
```

a normal append behavior would produce:

```
[1,2,3, [4,5]]
```

An append that does flattening will produce this:

```
[1,2,3,4,5]
```

But maybe I'm using the term incorrectly here.

Reply:
yes, I got this. I just wasn't sure if it was clear enough but probably is.

Reply:
I reverted the change